std/digest/xxh.d

```diff
@@ -0,0 +1,3352 @@
+}
+
+/*!
+ * @internal
+ * @brief The implementation for @ref XXH64().
+ *
+ * @param input , len , seed Directly passed from @ref XXH64().
+ * @param align Whether @p input is aligned.
+ * @return The calculated hash.
+ */
+private xxh_u64 XXH64_endian_align(const(xxh_u8)* input, size_t len,
+        xxh_u64 seed, XXH_alignment align_) @trusted pure nothrow @nogc
+{
+    xxh_u64 h64;
+    if (input == null)
+        assert(len == 0, "len != 0");
+
+    if (len >= 32)
+    {
+        const xxh_u8* bEnd = input + len;
+        const xxh_u8* limit = bEnd - 31;
+        xxh_u64 v1 = seed + XXH_PRIME64_1 + XXH_PRIME64_2;
+        xxh_u64 v2 = seed + XXH_PRIME64_2;
+        xxh_u64 v3 = seed + 0;
+        xxh_u64 v4 = seed - XXH_PRIME64_1;
+
+        do
+        {
+            v1 = XXH64_round(v1, XXH_get64bits(input, align_));
+            input += 8;
+            v2 = XXH64_round(v2, XXH_get64bits(input, align_));
+            input += 8;
+            v3 = XXH64_round(v3, XXH_get64bits(input, align_));
+            input += 8;
+            v4 = XXH64_round(v4, XXH_get64bits(input, align_));
+            input += 8;
+        }
+        while (input < limit);
+
+        h64 = XXH_rotl64(v1, 1) + XXH_rotl64(v2, 7) + XXH_rotl64(v3, 12) + XXH_rotl64(v4, 18);
+        h64 = XXH64_mergeRound(h64, v1);
+        h64 = XXH64_mergeRound(h64, v2);
+        h64 = XXH64_mergeRound(h64, v3);
+        h64 = XXH64_mergeRound(h64, v4);
+
+    }
+    else
+    {
+        h64 = seed + XXH_PRIME64_5;
+    }
+
+    h64 += cast(xxh_u64) len;
+
+    return XXH64_finalize(h64, input, len, align_);
+}
+
+XXH64_hash_t XXH64(const void* input, size_t len, XXH64_hash_t seed) @trusted pure nothrow @nogc
+{
+    static if (!XXH_NO_STREAM && XXH_SIZE_OPT >= 2)
+    {
+        /* Simple version, good for code maintenance, but unfortunately slow for small inputs */
+        XXH64_state_t state;
+        XXH64_reset(&state, seed);
+        XXH64_update(&state, cast(const(xxh_u8)*) input, len);
+        return XXH64_digest(&state);
+    }
+    else
+    {
+        if (XXH_FORCE_ALIGN_CHECK)
+        {
+            if (((cast(size_t) input) & 7) == 0)
+            { /* Input is aligned, let's leverage the speed advantage */
+                return XXH64_endian_align(cast(const(xxh_u8)*) input, len,
+                        seed, XXH_alignment.XXH_aligned);
+            }
+        }
+
+        return XXH64_endian_align(cast(const(xxh_u8)*) input, len, seed,
+                XXH_alignment.XXH_unaligned);
+    }
+}
+
+XXH64_state_t* XXH64_createState() @trusted pure nothrow @nogc
+{
+    import core.memory : pureMalloc;
+
+    return cast(XXH64_state_t*) pureMalloc(XXH64_state_t.sizeof);
+
+}
+
+XXH_errorcode XXH64_freeState(XXH64_state_t* statePtr) @trusted pure nothrow @nogc
+{
+    import core.memory : pureFree;
+
+    pureFree(statePtr);
+    return XXH_errorcode.XXH_OK;
+}
+
+void XXH64_copyState(XXH64_state_t* dstState, const XXH64_state_t* srcState) @trusted pure nothrow @nogc
```

Comment:
Probably replacing this with `emplace`

Reply:
Ok, I will try that.

Reply:
Dropped interface. It's not needed for operation. D can copy structures without such wrappers.